### PR TITLE
added ContextParser.reset()

### PR DESCRIPTION
**Pull request by adon-at-work**
- to reset internal states
- use case: https://github.com/yahoo/html-purify/blob/handle-unbalanced-tags/src/html-purify.js#L152

### CODE REVIEW

**Comment by adon-at-work** [comment]
@yukinying and @maditya , please review.

File: src/context-parser.js [hunk 1]
```diff
@@ -297,6 +308,8 @@ FastParser.prototype.lookupChar = function(ch) {
 
 /**
  * @function FastParser#contextualize
+ *
+ * @param {string} input - the input stream
```
Comment (adon-at-work) [inline src/context-parser.js]: btw, the prototype of `FastParser.contextualize()` is different from `Parser.contextualize()`. the one with Parser returns the modified (canonicalized) input, whereas that of FastParser never returns.
c.c. @neraliu

**Comment by yukinying** [comment]
+1

**Comment by maditya** [comment]
:+1:

**Comment by adon-at-work** [comment]
@yukinying / @maditya. merge it then. :)

**Pull request merged**